en/docs/security-guidelines/secure-engineering-guidelines/dynamic-analysis-with-owasp-zap.md

```diff
@@ -16,199 +16,199 @@ This document provides details of all necessary steps for configuring the OWASP
 
-This is helpful for testing a specific feature. For example, if we want to identify the possible issues in the user creation flow, we can create a new user in Management Console while ZAP traces all actions through the proxy. 
+This helps test a specific feature. For example, if we want to identify the possible issues in the user creation flow, we can create a new user in Management Console while ZAP traces all actions through the proxy. 
 
-![Placeholder]({{#base_path#}}/assets/images/secure-coding-guidelines/zap-10.png){ .post-image }
+![Placeholder](../../assets/images/secure-coding-guidelines/zap-10.png)
 
 On the URLs ZAP discovered, it can perform attacks to find possible issues. We can improve the coverage of the scan by manually performing all actions in the browser (i.e in Management Console) so that ZAP discovers each flow that it can try attacking. 
 
 You can also automate this by having selenium scripts for all UI actions. After setting the ZAP to act as the proxy, you can run selenium scripts so that in the browser, it automatically plays the actions you would do manually. For every action, ZAP will discover the URLs. 
 
 
 ### Removing Unnecessary Sites from Scan
-When the ZAP is acting as the proxy, all the URLs that the browser calls will be traced under the Sites in ZAP. We need to remove external websites and select only the WSO2 Server’s scope for scan. 
+When the ZAP is acting as the proxy, all the URLs that the browser calls will be traced under the Sites in ZAP. We need to remove external websites and select only the WSO2 Server’s scope for the scan. 
 
-Right click on the **Site** that should be included in the scan and select **Include in Context** &rarr; **New Context** .
+Right-click the **Site** that should be included in the scan and select **Include in Context** &rarr; **New Context** .
 
-![Placeholder]({{#base_path#}}/assets/images/secure-coding-guidelines/zap-11.png){ .post-image }
+![Placeholder](../../assets/images/secure-coding-guidelines/zap-11.png)
 
-Then it will show the **Session Properties** window with the regex for including the URL patterns in the scan. Provide a name for the context so that we can identify the site uniquely when we have multiple sites added as different contexts. (For example when you are testing a product like WSO2 API Manager, the Management Console, API Store and API Publisher can be three major sites where you can add each as a different context)
+Then it will show the **Session Properties** window with the regex for including the URL patterns in the scan. Provide a name for the context so that we can identify the site uniquely when we have multiple sites added as different contexts. (For example, when you are testing a product like WSO2 API Manager, the Management Console, API Store and API Publisher can be three major sites where you can add each as a different context)
```

Comment:
```suggestion
Then it will show the **Session Properties** window with the regex for including the URL patterns in the scan. Provide a name for the context so that we can identify the site uniquely when we have multiple sites added as different contexts. For example, when you are testing a product like WSO2 API Manager, the Management Console, API Store, and API Publisher can be three major sites where you can add each as a different context.
```